components/Survey.js

```diff
@@ -0,0 +1,55 @@
+const React = require('react');
+const Rating = require('./Rating');
+
+function random(array) {
+  return array[Math.floor(Math.random() * array.length)];
+}
+
+class Survey extends React.Component {
+
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      rating: 0
+    };
+
+    this.increaseRating = this.increaseRating.bind(this);
+    this.maintainRating = this.maintainRating.bind(this);
+    this.decreaseRating = this.decreaseRating.bind(this);
+  }
+
+  increaseRating() {
+    this.setState({ rating: this.state.rating + 1 });
+  }
+
+  decreaseRating() {
+    this.setState({ rating: this.state.rating - 1 });
+  }
+
+  maintainRating() {
```

Comment:
Yep, you're right. I think a cleaner way of doing this is using `this.forceUpdate()` though, setting the state to the same thing feels like an antipattern.